template/steps/prepare_datasets.py

```diff
@@ -33,12 +33,16 @@ def prepare_data(
 
     cleanup_gpu_memory(force=True)
 
-    log_model_metadata(
-        {
-            "system_prompt": system_prompt,
-            "base_model_id": base_model_id,
-        }
-    )
+    context = get_step_context()
+    if context.model:
+        log_metadata(
+            metadata={
+                "system_prompt": system_prompt,
+                "base_model_id": base_model_id,
+            },
+            model_name=context.model.name,
+            model_version=context.model.version,
+        )
```

Comment:
This looks like a quite complicated code piece vs what we had before, just saying 🙂